### PR TITLE
[ZEPPELIN-3450] Number sorting issue

### PR DESCRIPTION
**Pull request by sameer79**
### What is this PR for?
 Sorting value with types [NUMBER,STRING,DATE].

 ### What type of PR is it?
 [Bug Fix]

 ### What is the Jira issue?
 * [ZEPPELIN-3450](https://issues.apache.org/jira/browse/ZEPPELIN-3450)

 ### How should this be tested?
 First select the type of sorting, then click the sort asc/desc.

### Questions:
* Does the licenses files need update?
* Is there breaking changes for older versions?
* Does this needs documentation?

### CODE REVIEW

File: zeppelin-web/src/app/visualization/builtins/visualization-table.js [hunk 1]
```diff
@@ -120,6 +125,18 @@ export default class TableVisualization extends Visualization {
           `,
           minWidth: this.getColumnMinWidth(colName),
           width: '*',
+          sortingAlgorithm: function(a, b, row1, row2, sortType, gridCol) {
+            const colType = gridCol.colDef.type.toLowerCase();
+            if (colType === 'number') {
```
Comment (r-kamath) [inline zeppelin-web/src/app/visualization/builtins/visualization-table.js]: use TableColumnType.STRING
See TableColumnType def in `zeppelin-web/src/app/visualization/builtins/visualization-util.js`

Comment (sameer79) [reply zeppelin-web/src/app/visualization/builtins/visualization-table.js]: @r-kamath  address the comment and updated the PR as well.

**Comment by r-kamath** [comment]
will merge this if there are no more discussions

**Comment by Tagar** [comment]
Thanks for fixing this - I just had a chance to test and confirm it fixes number sorting.
PR title reads "Number sorting issue", but code change seems tries to address String and Date sorting issues too?

I see String sorting is still not correct even with this PR applied:
image

'Abba' should be at the top. It seems sort strings while ignoring their case?

**Comment by zjffdu** [comment]
Is it possible to have unit test for this ?

**Comment by sameer79** [comment]
@Tagar  Thanks for your comments - please provide the environment details.

**Comment by Tagar** [comment]
@sameer79 I was using latest Zeppelin from master snapshot. Chrome browser. 
Can you reproduce the same sorting issue for strings? 

Here's the code I used:
```python
%pyspark

print sorted(['abba2','abba','abba','Abba','abba','aBBa' ])

z.show(spark.sql("""
                select 1,2,0,   '2014-05-01', 'abba2' 
    union all   select 1,3,-1,  '2014-05-01', 'abba' 
    union all   select 1,5,3,   '2014-12-01', 'abba' 
    union all   select 1,9,-3,  '2014-05-01', 'Abba' 
    union all   select 1,2,-10, '2014-05-01', 'abba' 
    union all   select 2,7,3,   '2014-05-01', 'aBBa' 
"""))

```

**Comment by Tagar** [comment]
@sameer79 I refreshed my Zeppelin to latest master snapshot. 
Can't reproduce any longer the issue I specified on the screenshot above.
Please disregard my earlier comment. 
LGTM.
This would be great if this fix can make into 0.8 release.

**Comment by sameer79** [comment]
@Tagar Your welcome and thanks for your comments.

**Comment by prabhjyotsingh** [comment]
Merging this to master and branch-0.8 if no more discussion.

**Pull request closed without merging**